client/app/cloud/project/add/cloud-project-add.controller.js

```diff
@@ -61,89 +61,61 @@ angular.module("managerApp").controller("CloudProjectAddCtrl",
-                                }).$promise.then(function (contract) {
-                                    contract.id = contractId;
-                                    self.data.agreements.push(contract);
-                                }));
-                            });
-                        }
+                return CloudProjectAddService.orderCloudProject(self.model.description, self.model.voucher)
+                    .then(response => {
+                        self.data.activeOrder = response;
 
-                        // for multi-project: say to user that there are contracts to sign
-                        if (!self.loaders.init) {
-                            Toast.info($translate.instant("cpa_error_contracts_tosign"));
+                        if (response.prices.withTax.value) {
+                            $window.open(response.url);
                         }
 
-                        return $q.all(queue);
+                        self.pollOrder(response.orderId);
+                    })
+                    .catch(() => {
+                        self.data.activeOrder = undefined;
+                        Toast.error($translate.instant("cpa_error"));
+                    })
+                    .finally(() => {
+                        self.loaders.creating = false;
+                    });
+        };
 
-                    // case "validationPending":
-                    default:
-                        return;
-                    }
-                });
+        this.pollOrder = function (orderId) {
+            const cloudOrder = {};
+            if (self.data.poller) {
+                self.data.poller.kill();
+            }
 
-            })["catch"](function (err) {
-                if (err && err.status) {
-                    switch (err.status) {
-                    case 400:
-                        return Toast.error($translate.instant("cpa_error_invalid_paymentmean"));
-                    case 409:
-                        return Toast.error($translate.instant("cpa_error_over_quota"));
-                    default:
-                        return Toast.error($translate.instant("cpa_error") + (err.data && err.data.message ? " (" + err.data.message + ")" : ""));
-                    }
-                }
-            })["finally"](function () {
-                self.loaders.creating = false;
+            self.data.poller = CloudPoll.poll({
+                item: cloudOrder,
+                pollFunction: () => CloudProjectAddService.getCloudProjectOrder(orderId).then(response => {
+                        return response || {};
+                    }),
+                stopCondition: item => item.status === "delivered",
+                interval: 3000
             });
-        };
+            
+            self.data.poller.$promise
+                .then(() => {
+                    self.data.projectReady = true;
+                    return $timeout(() => {
```

Comment:
why there is a timeout ?

Reply:
because we want to display the animation "your project is ready" (green circle with a check mark) before redirecting.